src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp

```diff
@@ -51,13 +51,18 @@ bool ParserJSONPathMemberAccess::parseImpl(Pos & pos, ASTPtr & node, Expected &
         }
         const auto * last_begin = *pos->begin == '.' ? pos->begin + 1 : pos->begin;
         const auto * last_end = pos->end;
+        const String numberic_member_name(last_begin, last_end);
```

Comment:
move this into the `else if` block, maybe better

Reply:
done